content/blog/end-to-end-easy-to-use-pipeline-for-training-a-model-on-medmnist-v2-using-hpe-machine-learning-development-environment-flask.md

```diff
@@ -0,0 +1,514 @@
+* During training – If a trial fails on epoch 9 and the last checkpoint was saved during epoch 1, Determined will save yet another checkpoint at epoch 9, making this a very efficient system. 
+* Upon completion of the trial. 
+
+Additional checkpoint configuration settings can be modified to make this even more customizable. Checkpoints can be easily examined through the WebUI even after an experiment completes, and downloaded easily through the Determined APIs: 
+
+![](/img/screenshot3.png)
+
+#### Hyperparameter search 
+
+Hyperparameter search refers to the process of searching for the optimal configuration settings for your machine learning model – for example, searching for the optimal learning rate or convolution sizes in a convolutional neural network. The original training script does not implement hyperparameter search. This is not surprising, as hyperparameter search is yet another heavy lift for a researcher who wants to focus on experimentation, not the grunt work of setting up hyperparameter search.  
+
+With Determined, configuring hyperparameter search is easy. Defining hyperparameters, either static or in ranges, is easy through experiment configuration: 
+
+```yaml
+hyperparameters:
+    global_batch_size: 128
+    data_flag: pathmnist
+    dataset_name: "pathmnist.npz"
+    model_flag: resnet18
+    lr: 0.001
+    gamma: 0.1
+    resize: True
+    task: "multi-class"
+    num_epochs: 15
+```
+
+In this case, the “hyperparameters” feature can also be used to switch between datasets and types of training tasks if configured appropriately in the training script – which is neat! 
+
+#### Collaboration 
+
+Data scientists and researchers rarely work alone, especially in today’s data-driven technological boom. Sharing experiment results across a large team and sharing a single GPU cluster is not traditionally straightforward, but HPE Machine Learning Development Environment makes for a much better experience: 
+
+#### Resource Management 
+
+At the enterprise level, HPE Machine Learning Development Environment automatically scales up and down workloads depending on priority and resource availability. For example, if a researcher is running a hefty training job and is utilizing all 10 GPUs on a shared cluster, but their colleague needs two of them for a higher priority smaller job, HPE Machine Learning Development Environment can temporarily scale back the larger job utilizing all 10 GPUs down to using only 8, leaving room for the smaller training job.  
+
+#### Model Sharing and Reproducibility 
+
+The HPE Machine Learning Development Environment WebUI makes it easy to track experiments and see which model configurations resulted in particular results, across a team. This makes reproducibility easy, something that is of utmost importance when developing models for a use case like predicting cancer in biomedical images.  
+
+### Appendix: Building an E2E solution 
```

Comment:
Hmmm... So, this is really where you get into the meat of what you'd like to share with your reader.
I'd like you to think about this and consider some alternatives:

1. It might be possible to break up this blog post into two - with the first focused on all the goodness that the HPE Machine Learning Development Environment gives users (more of a "marketing" post) and the second one (more of a "technical post") focused on the experiment. If you were to do this, you would need to take some of the introduction material out of the first and use it for the introduction of the second. And, you would need to make sure you properly linked places between the two blog posts (easily done and we can help there).
2. Another alternative is to use your headers a little differently and not relegate your "solution build" to an Appendix.  This "Appendix" section is really what I had been waiting for all along as a reader... this is the "Let's get started!" piece. To use the headers a little differently, I'd recommend 

- drop the use of the term "Appendix"
- Make sure that the Introduction is in the largest header font
- Add one line below it that says "In this section, I'm going to cover what you need to know up front before you get started with the model training"
-  then use each of the headers that you have, at the same weight (except maybe the ones below "Collaboration", if they are indeed a part of "Collaboration". In that case, you'd make those two smaller.
- Then for this "Building an E2E solution", you might want to change it to "Let's build an E2E solution" and keep it the font the same weight as "Introduction"
- Place one line just under that and above "Step 1..." that says something like "Now that you have a good background on the HPE Machine Learning Development Environment, l'll show you how you can build an E2E solution for your model training."

I hope that helps. The way you have it now is "okay". But organizing it a little more will definitely help the reader follow along with what you're trying to convey.